src/QueryBuilder/QueryBuilder/helpers/valueBuilder.js

```diff
@@ -2,9 +2,16 @@ import { dayjs } from '@folio/stripes/components';
 import { DATA_TYPES } from '../../../constants/dataTypes';
 import { OPERATORS, OPERATORS_GROUPS_NAME } from '../../../constants/operators';
 import { getOperatorType } from './selectOptions';
+import { findLabelByValue } from '../../ResultViewer/utils';
 
 export const getCommaSeparatedStr = (arr) => {
-  const str = arr?.map(el => `"${el?.value}"`).join(',');
+  const str = arr?.map(el => {
+    if (/^(_custom_field|opt_)/.test(el.value)) {
+      return `"${el?.label}"`;
+    }
+
+    return `"${el?.value}"`;
+  }).join(',');
```

Comment:
It would be nice to have a function header describing the shape of the arguments and how they are transformed, as well as the significance of the string that are hard-coded into the regex. What is `_custom_field`? What is `opt_`? Why are they special?